evalml/pipelines/components/transformers/preprocessing/polynomial_decomposer.py

```diff
@@ -0,0 +1,349 @@
+from sktime.forecasting.base._fh import ForecastingHorizon
+from statsmodels.tsa.seasonal import seasonal_decompose
+from statsmodels.tsa.tsatools import freq_to_period
+
+from evalml.pipelines.components.transformers.preprocessing import Decomposer
+from evalml.utils import import_or_raise, infer_feature_types
+
+
+class PolynomialDecomposer(Decomposer):
+    """Removes trends and seasonality from time series by fitting a polynomial and moving average to the data.
+
+    Scikit-learn's PolynomialForecaster is used to generate the additive trend portion of the target data. A polynomial
+        will be fit to the data during fit.  That additive polynomial trend will be removed during fit so that statsmodel's
+        seasonal_decompose can determine the addititve seasonality of the data by using rolling averages over the series'
+        inferred periodicity.
+
+        For example, daily time series data will generate rolling averages over the first week of data, normalize
+        out the mean and return those 7 averages repeated over the entire length of the given series.  Those seven
+        averages, repeated as many times as necessary to match the length of the given target data, will be used
+        as the seasonal signal of the data.
+
+    Args:
+        time_index (str): Specifies the name of the column in X that provides the datetime objects. Defaults to None.
+        degree (int): Degree for the polynomial. If 1, linear model is fit to the data.
+            If 2, quadratic model is fit, etc. Defaults to 1.
+        random_seed (int): Seed for the random number generator. Defaults to 0.
+    """
+
+    name = "Polynomial Decomposer"
+    hyperparameter_ranges = {"degree": Integer(1, 3)}
+    """{
+        "degree": Integer(1, 3)
+    }"""
+    modifies_features = False
+    modifies_target = True
+
+    def __init__(
+        self, time_index: str = None, degree: int = 1, random_seed: int = 0, **kwargs
+    ):
+        if not isinstance(degree, int):
+            if isinstance(degree, float) and degree.is_integer():
+                degree = int(degree)
+            else:
+                raise TypeError(
+                    f"Parameter Degree must be an integer!: Received {type(degree).__name__}",
+                )
+
+        params = {"degree": degree}
+        params.update(kwargs)
+        error_msg = "sktime is not installed. Please install using 'pip install sktime'"
+
+        trend = import_or_raise("sktime.forecasting.trend", error_msg=error_msg)
+        detrend = import_or_raise(
+            "sktime.transformations.series.detrend",
+            error_msg=error_msg,
+        )
+
+        decomposer = detrend.Detrender(trend.PolynomialTrendForecaster(degree=degree))
+
+        params["time_index"] = time_index
+
+        super().__init__(
+            parameters=params,
+            component_obj=decomposer,
+            random_seed=random_seed,
+        )
+
+    def _set_time_index(self, X, y):
```

Comment:
It might...I was thinking about doing that separately though, so I could spend the time figuring out what other components could use it.  It might be appropriate to move it up the inheritance ladder?

Reply:
I filed this : https://github.com/alteryx/evalml/issues/3716